Sources/IntegerUtilities/Rounding.swift

```diff
@@ -0,0 +1,65 @@
+//===--- Rounding.swift ---------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// A rule that defines how to select one of the two representable results
+/// closest to a given value.
+public enum RoundingRule {
```

Comment:
This feels a little inconsistent.

`…, rounding: .down` ✅ 
`..., rounding: .stochastic` should this be "stochastically"?
`..., rounding: .trap` in what circumstances does it trap? (I realize this is clear if you think about it for a moment, but maybe we can make it obvious at the point of use?)

Related, do we want `rounding` or do we want `rounded`? With the name `shifted` we're describing the resulting value, rather than the action, but with `rounding` we're describing the action.

Reply:
Yeah, I'm not super-delighted with "trap" either. Still trying to come up with a better name for it.

Reply:
`trapIfUnrepresentable`?

Reply:
`.exactOrTrap` maybe?

Reply:
Maybe just `.exact`, though I would like to mention "trap" in the name. I like the symmetry of `exact` with the `exactly:` inits from the stdlib.

Reply:
I like .exactOrTrap  or something similar.  "Trap" is kind of a specialist word, though, so maybe .exactOrFatal?  .exactOrAssert?  .mustBeExact?  I think it makes sense to have this one be linguistically distinct, so .onlyExact or .mustBeExact or .remainderMustBeZero or ...

Reply:
If this was implemented in the standard library, would the new cases be added to [`FloatingPointRoundingRule`](https://developer.apple.com/documentation/swift/floatingpointroundingrule), or should the new enum be renamed to `IntegerRoundingRule` instead?

Reply:
I would like to deprecate `FloatingPointRoundingRule` eventually and replace it with `RoundingRule`. Exactly what that migration path looks like isn't totally clear, but I think it's the direction that we'll take if/when these show up in the stdlib.